api/datasets.go

```diff
@@ -2,13 +2,18 @@ package api
 
 import (
 	"fmt"
+	"net/http"
 
 	"github.com/application-research/delta-dm/core"
 	db "github.com/application-research/delta-dm/db"
 	"github.com/application-research/delta-dm/util"
 	"github.com/labstack/echo/v4"
 )
 
+type DatasetPutBody struct {
```

Comment:
I think we should also allow the repl. quota and duration to be changed here too - just a generic PUT endpoint that can modify everything

Reply:
Added repl. quota / deall duration to the update object. I've also added basic validation around these fields.